ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java

```diff
@@ -120,21 +122,80 @@ private double getRemedialActionResultValue(RemedialActionResult remedialActionR
         throw new FaraoException("Range action type of " + remedialActionElementResult.getId() + " is not supported yet");
     }
 
-    private boolean sameRemedialActionResultLists(List<RemedialActionResult> firstList, List<RemedialActionResult> secondList) {
-        if (firstList.size() != secondList.size()) {
-            return false;
+    private RemedialActionResult combineRAResults(RemedialActionResult oldRemedialActionResult, RemedialActionResult newRemedialActionResult) {
+        //mapping the old RA elements to their IDs
+        Map<String, RemedialActionElementResult> oldRAResultElementMap = new HashMap<>();
+        for (RemedialActionElementResult remedialActionElementResult : oldRemedialActionResult.getRemedialActionElementResults()) {
+            oldRAResultElementMap.put(remedialActionElementResult.getId(), remedialActionElementResult);
+        }
+        //initializing an empty RA element list which will contain the combination of the elements from both RAs
+        List<RemedialActionElementResult> combinedRAResultElementList = new ArrayList<>();
+        /*
+        For each RA element of the second RA, we create a new RA element combining that RA element and the corresponding
+          one from the first RA (as long as they dont cancel each other out)
+        */
+        for (RemedialActionElementResult remedialActionElementResult : newRemedialActionResult.getRemedialActionElementResults()) {
+            if (remedialActionElementResult instanceof PstElementResult) {
+                PstElementResult newPstElementResult = (PstElementResult) remedialActionElementResult;
+                PstElementResult oldPstElementResult = (PstElementResult) oldRAResultElementMap.get(remedialActionElementResult.getId());
+                double totalChange = newPstElementResult.getPostOptimisationAngle() - oldPstElementResult.getPreOptimisationAngle();
+                if (totalChange > 0.0001) {
```

Comment:
You can define a static final variable to define this minimum threshold

Reply:
ok